tokio/src/runtime/mod.rs

```diff
@@ -545,7 +558,11 @@ impl Runtime {
     pub fn shutdown_timeout(mut self, duration: Duration) {
         // Wakeup and shutdown all the worker threads
         self.handle.spawner.shutdown();
-        self.blocking_pool.shutdown(Some(duration));
+
+        // TODO: this is likely incorrect, we need to find some way to synchronize shutting down.
```

Comment:
are there things we should be testing around this? what's up here?

Reply:
I have no clue but I wanted to dig into this in its own PR, this right now seems to pass all our tests but I want to dig in deeper. So punting on this for this PR.

Reply:
When you remove the `Arc`, it should be fine. It's "correct" because the fn takes `&mut self`.